sign/impl.go

```diff
@@ -47,55 +48,96 @@ type defaultImpl struct{}
 type impl interface {
-	VerifyFileInternal(ctx context.Context, ko options.KeyOpts, outputSignature, outputCertificate, path string) error
-	VerifyImageInternal(ctx context.Context, keyPath string, images []string) (*SignedObject, error)
-	SignImageInternal(ro options.RootOptions, ko options.KeyOpts, regOpts options.RegistryOptions,
-		annotations map[string]interface{}, imgs []string, certPath string, upload bool,
-		outputSignature string, outputCertificate string, payloadPath string, force bool,
-		recursive bool, attachment string, noTlogUpload bool) error
+	VerifyFileInternal(ctx context.Context, ko options.KeyOpts, certOpts options.CertVerifyOptions, outputSignature, path string) error
+	VerifyImageInternal(ctx context.Context, certOpts options.CertVerifyOptions, keyPath string, images []string, ignoreTLog bool) (*SignedObject, error)
+	SignImageInternal(ro options.RootOptions, ko options.KeyOpts, signOpts options.SignOptions, imgs []string) error
 	SignFileInternal(ro options.RootOptions, ko options.KeyOpts, regOpts options.RegistryOptions,
-		payloadPath string, b64 bool, outputSignature string, outputCertificate string) error
+		payloadPath string, b64 bool, outputSignature string, outputCertificate string, tlogUpload bool) error
 	Setenv(string, string) error
 	EnvDefault(string, string) string
 	TokenFromProviders(context.Context, *logrus.Logger) (string, error)
 	FileExists(string) bool
 	ParseReference(string, ...name.Option) (name.Reference, error)
-	FindTLogEntriesByPayload(ctx context.Context, rClient *client.Rekor, blobBytes []byte) ([]string, error)
+	FindTLogEntriesByPayload(ctx context.Context, rClient *client.Rekor, b64Sig string, blobBytes []byte, pubKey []byte) ([]models.LogEntryAnon, error)
+	FindTlogEntry(ctx context.Context, rClient *client.Rekor, b64Sig string, blobBytes []byte, pubKey []byte) ([]models.LogEntryAnon, error)
 	Digest(ref string, opt ...crane.Option) (string, error)
 	PayloadBytes(blobRef string) ([]byte, error)
 	NewRekorClient(string) (*client.Rekor, error)
 	NewWithContext(context.Context, name.Registry, authn.Authenticator, http.RoundTripper, []string) (http.RoundTripper, error)
 	ImagesSigned(context.Context, *Signer, ...string) (*sync.Map, error)
 }
 
-func (*defaultImpl) VerifyFileInternal(ctx context.Context, ko options.KeyOpts, outputSignature, //nolint: gocritic
-	outputCertificate, path string,
+type VerifyBlobCmd struct {
+	options.KeyOpts
+	options.CertVerifyOptions
+	CertRef                      string
+	CertChain                    string
+	SigRef                       string
+	CertGithubWorkflowTrigger    string
+	CertGithubWorkflowSHA        string
+	CertGithubWorkflowName       string
+	CertGithubWorkflowRepository string
+	CertGithubWorkflowRef        string
+	IgnoreSCT                    bool
+	SCTRef                       string
+	Offline                      bool
+	IgnoreTlog                   bool
+}
```

Comment:
What is this type for?

Reply:
to be honest i dont know, maybe was just to me to see all the fields, removed :)